client/src/pages/poster/tasks/[taskid]/[bidderid]/bid-details.tsx

```diff
@@ -0,0 +1,35 @@
+import { useRouter } from "next/router";
+import React from "react";
+
+import { Button } from "@/components/ui/button";
+import Header from "@/components/ui/header";
+import PersonDetail from "@/components/ui/person-detail";
+
+export default function BidDetail() {
+  const router = useRouter();
+  if (router.query.bidder === undefined || router.query.taskid === undefined) {
+    return <>Loading...</>;
+  }
+  const bidder = JSON.parse(String(router.query.bidder));
```

Comment:
could you please explain what `router.query.bidder` is or do you mean `router.query.bidderid` instead?

Reply:
The bidder json string was obtained from last page task-details page. I won't do a get request from backend, by data you've alreay got from previous pages, sharing between different pages.
For exmaple, bio, name, profile data of bidder just obtain from last pages, there's no need to call api of get request.

Reply:
cool I got it! well done